app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -93,20 +84,40 @@ def test_connection(options = {})
       super(options)
     end
 
-    def new_vm(attr = {})
-      AzureRMCompute.new(sdk: sdk)
+    def new_vm(args = {})
+      return AzureRMCompute.new(sdk: sdk) if args.empty? || args[:image_id].nil?
+      opts = vm_instance_defaults.merge(args.to_h).deep_symbolize_keys
+      # convert rails nested_attributes into a plain hash
+      [:interfaces].each do |collection|
```

Comment:
Do we still need this generic code, or can we simplify it to:

``` ruby
nested_args = opts.delete(:interfaces_attributes)
opts[collection] = nested_attributes_for(:interfaces, nested_args) if nested_args
```

Reply:
done.